src/NuGet.Clients/PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -159,6 +159,27 @@ private static LoadingStatus AggregateLoadingStatus(IEnumerable<LoadingStatus> s
             return totalCount;
         }
 
+        public async Task<IReadOnlyList<IPackageSearchMetadata>> GetAllPackagesAsync(CancellationToken cancellationToken)
+        {
+            var packages = new List<IPackageSearchMetadata>();
+            ContinuationToken nextToken = null;
+            do
+            {
+                var searchResult = await SearchAsync(nextToken, cancellationToken);
+                while (searchResult.RefreshToken != null)
+                {
+                    searchResult = await _packageFeed.RefreshSearchAsync(searchResult.RefreshToken, cancellationToken);
+                }
+
+                nextToken = searchResult.NextToken;
+
+                packages.AddRange(searchResult.Items);
+
+            } while (nextToken != null);
```

Comment:
my change follows other style in this file - i'll update both in a separate checkin - there are a few other things - for example missing copyright headers in a couple of files.